tests/utils/test_eloop.c

```diff
@@ -0,0 +1,169 @@
+#define _GNU_SOURCE
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <string.h>
+#include <inttypes.h>
+#include <unistd.h>
+#include <setjmp.h>
+#include <stdint.h>
+#include <cmocka.h>
+
+#include "utils/log.h"
+#include "utils/eloop.h"
+#include "utils/sockctl.h"
+
+#define TMP_PFX "tdoms"
+#define TEST_ELOOP_PARAM "param"
+#define TEST_SEND_BUF_DATA "test"
+
+void test_eloop_sock_handler_read(int sock, void *eloop_ctx, void *sock_ctx) {
+  struct eloop_data *eloop = (struct eloop_data *)eloop_ctx;
+  char *sock_ctx_data = (char *)sock_ctx;
+
+  assert_non_null(eloop);
+  assert_string_equal(sock_ctx_data, TEST_ELOOP_PARAM);
+
+  struct client_address addr;
+  char read_buf[100];
+
+  os_memset(&addr, 0, sizeof(struct client_address));
+  addr.type = SOCKET_TYPE_DOMAIN;
+  read_socket_data(sock, read_buf, 100, &addr, 0);
+  assert_string_equal(read_buf, TEST_SEND_BUF_DATA);
+
+  eloop_terminate(eloop);
+}
+
+void test_eloop_sock_handler_unreg(int sock, void *eloop_ctx, void *sock_ctx) {
+  (void)sock;
+  (void)eloop_ctx;
+  (void)sock_ctx;
+}
+
+void test_eloop_timeout_handler(void *eloop_ctx, void *user_ctx) {
```

Comment:
Because it's a bit hard to see what this function does.

```suggestion
/**
 * @param eloop_ctx Pointer to struct eloop_data.
 * @param[out] user_ctx Pointer to string buffer.
 * @post @p user_ctx will be written with TEST_ELOOP_PARAM.
 */
void test_eloop_timeout_handler(void *eloop_ctx, void *user_ctx) {
```